Python/pystate.c

```diff
@@ -2345,8 +2338,9 @@ PyThreadState_SetAsyncExc(unsigned long id, PyObject *exc)
      * list of thread states we're traversing, so to prevent that we lock
      * head_mutex for the duration.
      */
-    HEAD_LOCK(runtime);
-    for (PyThreadState *tstate = interp->threads.head; tstate != NULL; tstate = tstate->next) {
+    INTERP_THREAD_LOCK(interp);
+    PyThreadState *list = PyInterpreterState_ThreadHead(interp);
+    for (PyThreadState *tstate = list; tstate != NULL; tstate = tstate->next) {
```

Comment:
There isn't any need for the intermediate variable:

```suggestion
    PyThreadState *tstate = PyInterpreterState_ThreadHead(interp);
    for (; tstate != NULL; tstate = tstate->next) {
```

I was also going to suggest introducing a macro of static inline function to use in place of the puiblic API `PyInterpreterState_ThreadHead()`:

```suggestion
    PyInterpreterState_ThreadHead(interp);
    for (PyThreadState *tstate = thread_head(interp); tstate != NULL; tstate = tstate->next) {
```

However, that probably isn't worth it.

FWIW, you'd probably be fine with dropping this change.  Think of pystate.c as the home for `PyInterpreterState` and code that accesses the struct fields directly isn't that big a deal.  That said, I do support being careful to consolidate the places that we use struct fields directly, for better maintainability.

Reply:
The same goes for the other places in pystate.c where you've switched to using `PyInterpreterState_ThreadHead()`.

Reply:
I'm made changes, Thank for your comment :)